internal/cmd/operator-sdk/generate/internal/relatedimages.go

```diff
@@ -83,6 +85,28 @@ func (c *relatedImageCollector) collectFromEnvironment(containerRef string, env
 	return nil
 }
 
+func (c *relatedImageCollector) collectFromBaseCSV(base *operatorsv1alpha1.ClusterServiceVersion) error {
+	if base == nil || base.Spec.RelatedImages == nil {
+		return nil
+	}
+
```

Comment:
We shouldn't be calling this function at all if there's no base CSV. The caller should own checking if it is present or not. Also, you don't need to nil check the related images since they are in a slice you can safely iterate over a `nil` slice so long as its typed:

```go
x := len(nil) // will panic
y := len([]int(nil)) // won't panic
```